### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by celiot-IDM**
Starting a mini-project to add contribution guidelines to PACE-HRH

### CODE REVIEW

**Comment by celiot-IDM** [comment]
Now that Ryan has finished a first sizeable PR, I'm going to start adding him as a reviewer on my own PRs so he continues to gain familiarity with the codebase.

**Comment by RHull-IDM** [comment]
I see that you fixed a file to match the new style requirements. Do we intend to clean up all the source files as part of this PR?

**Comment by celiot-IDM** [comment]
> I see that you fixed a file to match the new style requirements. Do we intend to clean up all the source files as part of this PR?

Great question! I think we should pull together a deliberate plan to fix everything up. So not everything right now ... but let's put a plan in place.

**Review by MeWu-IDM — COMMENTED** [review]
Is this tool https://github.com/r-lib/styler compatible with the style defined?

**Comment by celiot-IDM** [comment]
> Is this tool https://github.com/r-lib/styler compatible with the style defined?

Styler is a great alternative to the RStudio Code/Reformat Code command to get code quickly to a good place.

But that said, lintr should be our definitive reference tool because we can eventually automate lintr::lint() checking into a GH Action, if we so choose.

**Comment by MeWu-IDM** [comment]
template for GA https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/287 , maybe we can customize based on above info

**Comment by celiot-IDM** [comment]
Almost finished! With these commits the code in the R directory (not the unit test code) passes linting.

**Comment by celiot-IDM** [comment]
@MeWu-IDM  and @RHull-IDM :

Ready to review. I apologize for touching pretty much every source file. https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/286/commits/a93e611d523a5477e712b56e4d8b1f7ca2258cbd is the only commit that makes a substantial code change. Note that I added a .lint file for lint configuration.

**Review by MeWu-IDM — COMMENTED** [review]
- I saw warning from the check:, I think you should add these files to the  .Rbuildignore
 ```
 .lintr
  vignettes/.gitignore
These were most likely included in error. See section ‘Package
structure’ in the ‘Writing R Extensions’ manual.
```
- When I run lint check on the package, I am still seeing a few issues
```
Error: Not lint free
R\pace_exp_control.R:53:6: style: [indentation_linter] Indentation should be 8 spaces but is 6 spaces.
      is.null(BVE$seasonalityCurves) ||
     ^~~

R\pace_prevalence_matrix.R:100:10: style: [indentation_linter] Indentation should be 8 spaces but is 10 spaces.
          tasks$ServiceCat == "Tuberculosis" | tasks$ServiceCat == "HIV"
        ~^

R\pace_table_schemas.R:135:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
    )
  ~^

tests\testthat\setup.R:40:1: style: [object_name_linter] Variable and function name style should match camelCase, CamelCase or symbols.
local_vars <- function(vars = vector(), envir = parent.frame(), .local_envir = parent.frame()){
^~~~~~~~~~

tests\testthat\setup.R:40:65: style: [object_name_linter] Variable and function name style should match camelCase, CamelCase or symb
```

**Comment by celiot-IDM** [comment]
> * I saw warning from the check:, I think you should add these files to the  .Rbuildignore
> 
> ```
> .lintr
>  vignettes/.gitignore
> These were most likely included in error. See section ‘Package
> structure’ in the ‘Writing R Extensions’ manual.
> ```
> 
> * When I run lint check on the package, I am still seeing a few issues
> 
> ```
> Error: Not lint free
> R\pace_exp_control.R:53:6: style: [indentation_linter] Indentation should be 8 spaces but is 6 spaces.
>       is.null(BVE$seasonalityCurves) ||
>      ^~~
> 
> R\pace_prevalence_matrix.R:100:10: style: [indentation_linter] Indentation should be 8 spaces but is 10 spaces.
>           tasks$ServiceCat == "Tuberculosis" | tasks$ServiceCat == "HIV"
>         ~^
> 
> R\pace_table_schemas.R:135:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
>     )
>   ~^
> 
> tests\testthat\setup.R:40:1: style: [object_name_linter] Variable and function name style should match camelCase, CamelCase or symbols.
> local_vars <- function(vars = vector(), envir = parent.frame(), .local_envir = parent.frame()){
> ^~~~~~~~~~
> 
> tests\testthat\setup.R:40:65: style: [object_name_linter] Variable and function name style should match camelCase, CamelCase or symb
> ```

Files added to .gitignore (practicing my Perl-style regex)

Turns out my version of lintr wasn't up-to-date, and didn't include the indentation_linter(). Frustratingly, styler was up-to-date, but fails sometimes to style in a way that passes the default indentation_linter() settings. Indentation problems are easy to fix, so I stopped trying to configure indentation_linter() to line up with styler.

@MeWu-IDM : I fixed up the files in the R directory, but I wasn't planning on styling the unit test code (tests\testthat\*). Can the build-time linting be restricted to the R directory?

**Comment by celiot-IDM** [comment]
@MeWu-IDM : Please investigate.

image

image

**Comment by MeWu-IDM** [comment]
@celiot-IDM 
This error is due to debian release 12.1 on July22
I have a temporary fix in  https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/287/files
I also add tests to exclude list 
I see this lint_check still fails in https://github.com/InstituteforDiseaseModeling/PACE-HRH/actions/runs/5767054196/job/15636111794#step:7:142
are these warning appear on your end?

**Comment by celiot-IDM** [comment]
> @celiot-IDM This error is due to debian release 12.1 on July22 I have a temporary fix in https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/287/files I also add tests to exclude list I see this lint_check still fails in https://github.com/InstituteforDiseaseModeling/PACE-HRH/actions/runs/5767054196/job/15636111794#step:7:142 are these warning appear on your end?

Those lint failures make no sense ... they imply that the package references list (eg. traceMessage function label) has disappeared. More likely a build problem than a lint problem.

**Comment by MeWu-IDM** [comment]
> > @celiot-IDM This error is due to debian release 12.1 on July22 I have a temporary fix in https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/287/files I also add tests to exclude list I see this lint_check still fails in https://github.com/InstituteforDiseaseModeling/PACE-HRH/actions/runs/5767054196/job/15636111794#step:7:142 are these warning appear on your end?
> 
> Those lint failures make no sense ... they imply that the package references list (eg. traceMessage function label) has disappeared. More likely a build problem than a lint problem.

Are you looking at the error link from my other PR? They are something like this:

Error: Not lint free
R/pace_cadre_config.R:8:40: warning: [object_usage_linter] no visible binding for global variable '.defaultCadreRolesSheet'
loadCadreRoles <- function(sheetName = .defaultCadreRolesSheet) {

**Comment by celiot-IDM** [comment]
> > > @celiot-IDM This error is due to debian release 12.1 on July22 I have a temporary fix in https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/287/files I also add tests to exclude list I see this lint_check still fails in https://github.com/InstituteforDiseaseModeling/PACE-HRH/actions/runs/5767054196/job/15636111794#step:7:142 are these warning appear on your end?
> > 
> > 
> > Those lint failures make no sense ... they imply that the package references list (eg. traceMessage function label) has disappeared. More likely a build problem than a lint problem.
> 
> Are you looking at the error link from my other PR? They are something like this:
> 
> Error: Not lint free R/pace_cadre_config.R:8:40: warning: [object_usage_linter] no visible binding for global variable '.defaultCadreRolesSheet' loadCadreRoles <- function(sheetName = .defaultCadreRolesSheet) {

Yep, we're looking at the same thing. This message is claiming that .defaultCadreRolesSheet isn't defined, but it is in pace_package_env.R:

image

Something else is going on in the build to lose track of global variable names.

**Comment by MeWu-IDM** [comment]
> > > > @celiot-IDM This error is due to debian release 12.1 on July22 I have a temporary fix in https://github.com/InstituteforDiseaseModeling/PACE-HRH/pull/287/files I also add tests to exclude list I see this lint_check still fails in https://github.com/InstituteforDiseaseModeling/PACE-HRH/actions/runs/5767054196/job/15636111794#step:7:142 are these warning appear on your end?
> > > 
> > > 
> > > Those lint failures make no sense ... they imply that the package references list (eg. traceMessage function label) has disappeared. More likely a build problem than a lint problem.
> > 
> > 
> > Are you looking at the error link from my other PR? They are something like this:
> > Error: Not lint free R/pace_cadre_config.R:8:40: warning: [object_usage_linter] no visible binding for global variable '.defaultCadreRolesSheet' loadCadreRoles <- function(sheetName = .defaultCadreRolesSheet) {
> 
> Yep, we're looking at the same thing. This message is claiming that .defaultCadreRolesSheet isn't defined, but it is in pace_package_env.R:
> 
> image
> 
> Something else is going on in the build to lose track of global variable names.

It might be this https://github.com/r-lib/lintr/issues/1137
I Will Check more on Monday

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+### Using styler and lintr
+
+The R packages [styler](https://styler.r-lib.org/) and [lintr](https://lintr.r-lib.org/) are the core tools for managing code style. We recommend installing both to your RStudio IDE.
+
+The __styler__ package re-formats code according to whatever style rules have been set up.
+
+The __lintr__ package checks for code style problems. 
+
+The linting configuration is controlled by the .lint file in the package root directory.
```
Comment (MeWu-IDM) [inline .github/CONTRIBUTING.md:57]: type ".lintr"

**Review by MeWu-IDM — APPROVED** [review]
please fix small typo in ReadMe then go ahead to merge this, good work!

**Pull request merged**